satellite/inspector/inspector.go

```
@@ -0,0 +1,184 @@
+// Copyright (C) 2019 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package inspector
+
+import (
+	"context"
+	"strconv"
+
+	"github.com/skyrings/skyring-common/tools/uuid"
+	"github.com/zeebo/errs"
+	"go.uber.org/zap"
+
+	monkit "gopkg.in/spacemonkeygo/monkit.v2"
+
+	"storj.io/storj/pkg/eestream"
+	"storj.io/storj/pkg/overlay"
+	"storj.io/storj/pkg/pb"
+	"storj.io/storj/pkg/pointerdb"
+	"storj.io/storj/pkg/storj"
+	"storj.io/storj/satellite/metainfo"
+)
+
+var (
+	mon = monkit.Package()
+	// Error wraps errors returned from Server struct methods
+	Error        = errs.Class("Endpoint error")
+	finalSegment = int64(-1)
+)
+
+// Endpoint for checking object and segment health
+type Endpoint struct {
+	pointerdb *pointerdb.Service
+	cache     *overlay.Cache
+	log       *zap.Logger
+}
+
+// NewEndpoint will initialize an Endpoint struct
+func NewEndpoint(log *zap.Logger, cache *overlay.Cache, pdb *pointerdb.Service) (*Endpoint, error) {
```

Do you need to return an error ?